nirc_ehr/resources/web/nirc_ehr/model/sources/ClinicalDefaults.js

```diff
@@ -4,43 +4,104 @@
  * Licensed under the Apache License, Version 2.0: http://www.apache.org/licenses/LICENSE-2.0
  */
 EHR.model.DataModelManager.registerMetadata('ClinicalDefaults', {
-    allQueries: {
-        performedby: {
-            defaultValue: LABKEY.Security.currentUser.displayName
-        },
-    },
     byQuery: {
         'study.treatment_order': {
             category: {
                 defaultValue: 'Clinical',
                 hidden: true
-            }
+            },
+            performedby: {
+                hidden: false,
+                allowBlank: false,
+                defaultValue: null,
+                lookup: {
+                    schemaName: 'ehr_lookups',
+                    queryName: 'veterinarians',
+                    keyColumn: 'UserId',
+                    displayColumn: 'DisplayName'
+                }
+            },
         },
         'study.drug': {
             category: {
                 defaultValue: 'Clinical',
                 hidden: true
+            },
+            performedby: {
+                hidden: false,
+                allowBlank: false,
+                defaultValue: null,
+                lookup: {
+                    schemaName: 'ehr_lookups',
+                    queryName: 'veterinarians',
+                    keyColumn: 'UserId',
+                    displayColumn: 'DisplayName'
+                }
             }
         },
-        'study.procedure': {
+        'study.prc': {
             category: {
                 defaultValue: 'Clinical',
                 hidden: true
-            }
+            },
         },
         'study.clinremarks': {
             category: {
                 defaultValue: 'Clinical',
                 hidden: true,
                 allowBlank: false
+            },
+            performedby: {
+                hidden: false,
+                defaultValue: LABKEY.Security.currentUser.id.toString(),
+                editorConfig: {
+                    store: {
+                        type: 'labkey-store',
+                        schemaName: 'core',
+
+                        // 'performedby' is a text field in the dataset and its lookup to the userid is an int field - this mismatch causes it to disappear
+                        // from the display when a value is selected from the dropdown even though the 'userid' value gets saved as a text (this behavior was only seen
+                        // in the form panel but not in the grid panel).
+                        // casting it as a varchar when loading the store fixes this issue.
+                        sql: 'SELECT CAST (UserId AS VARCHAR) AS UserId,DisplayName,FirstName,LastName FROM core.users',
+                        autoLoad: true
+                    }
+                }
```

Comment:
this config is duplicated multiple times. can this be moved to a shared place in a file that is injected by base form java class? 'TNPRC' has an example of global.js file?